pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -436,6 +436,16 @@ func (cb *ClusterBuilder) buildInboundClusterForPortOrUDS(clusterPort int, bind
 	}
 	localCluster := cb.buildDefaultCluster(clusterName, clusterType, localityLbEndpoints,
 		model.TrafficDirectionInbound, instance.ServicePort, instance.Service, allInstance)
+	if cb.hbone && clusterType == cluster.Cluster_ORIGINAL_DST {
+		localCluster.cluster.LbConfig = &cluster.Cluster_OriginalDstLbConfig_{
```

Comment:
@kyessenov I am not sure this is quite right (but maybe good enough for now...?). This sets the port correctly on the upstream request, but I think we will apply RBAC with an invalid port.

Should we modify connect_authority to allow taking port from the header as well?

Reply:
We could leverage connect_authority here since it's per CONNECT request. I couldn't make upstream_port_override work for me, seems like the cluster is used in more contexts than desired.

connect_authority has a port value in its config to override tunnel port. We could make a default value to be the same as authority port instead of 15008 since there is no tunnel on the upstream in the sidecar.

Reply:
In this context the cluster is `inbound|80||` or whatever so override to 80 seems to work, just not sure its the best idea since other filters may see the wrong port (logs, for example, show `0` as the port for this change).

> connect_authority has a port value in its config to override tunnel port. We could make a default value to be the same as authority port instead of 15008 since there is no tunnel on the upstream in the sidecar.

That makes sense to me